OpenKeychain/src/main/java/org/sufficientlysecure/keychain/provider/KeychainProvider.java

```diff
@@ -420,6 +422,16 @@ public Cursor query(@NonNull Uri uri, String[] projection, String selection, Str
                                     + " AND ( kA." + Keys.EXPIRY + " IS NULL OR kA." + Keys.EXPIRY
                                     + " >= " + new Date().getTime() / 1000 + " )"
                                     + ")" : "")
+                        + (plist.contains(KeyRings.HAS_AUTHENTICATE_PUBLIC) ?
```

Comment:
What makes this "public"? It will include secret keys as well, won't it? Is this distinction really necessary?

Reply:
Yes, I'm not too sure about the naming. I chose to keep to keep HAS_AUTHENTICATE as having secret keys available to keep in line with HAS_SIGN etc. An earlier version changed HAS_AUTHENTICATE to be inclusive and added HAS_AUTHENTICATE_SECRET, would that be more appropriate?

Reply:
Ugh, this whole thing is messier than I had in mind. I don't have a good answer, the `HAS_` things are already inconsistent as they are.

Reply:
No worries. I was thinking about making it all consistent by specialising into `HAS_*` / `HAS_*_SECRET` since I couldn't find a good reason for the naming as it is; I'll look into that later.